[18.0][MIG] base_tier_validation: Migration to 18.0

xaviedoanhduy opened the pull request:
take over PR: https://github.com/OCA/server-ux/pull/966

changes are follow up from old PR:

- fix the icon on systray 
- fix of not recalculating reviewCount when updating review requests (`TierReviewService`). don't use useState to calculate tierReviewCounter because the service will update `services["mail.store"]`
- replaced old-style string interpolation with `f-string `and placeholder formatting
- remove old jquery and replace with javascript available objects
- optimize multi-company rule
- replace deprecated output directives (`t-esc`)

etobella committed:
[ADD] Sequence check Digit (#1000)

* [ADD] Sequence check Digit

* Change of License to LGPL

* Code review

* Reviews

etobella committed:
[MIG] sequence_check_digit: Migration to 11.0

LoisRForgeFlow committed:
[9.0][ADD] base_tier_validation

LoisRForgeFlow committed:
fix: blocking unneded records

LoisRForgeFlow committed:
[9.0][IMP] base_tier_validation: tries automatically request validation and validate if possible.

LoisRForgeFlow committed:
[9.0][IMP] base_tier_validation: filter out reviews not pending

LoisRForgeFlow committed:
make possible to filter by validated records

LoisRForgeFlow committed:
allow to add exceptions for fields that can be written on under validation records

LoisRForgeFlow committed:
[9.0][IMP] base_tier_validation:
* able to restart validation
* sudo() not needed anymore

LoisRForgeFlow committed:
[10.0][MIG] base_tier_validation

LoisRForgeFlow committed:
[11.0][MIG] base_tier_validation

LoisRForgeFlow committed:
[11.0][IMP] base_tier_validation: add tests

oca-travis committed:
[UPD] Update sequence_check_digit.pot

JordiBForgeFlow committed:
add 'can_review' to tier.validation so that the buttons approve
and reject can be hidden according to this computed field.

etobella committed:
Translated using Weblate (Español (España))

Currently translated at 100,0% (13 of 13 strings)

Translation: server-ux-11.0/server-ux-11.0-sequence_check_digit
Translate-URL: https://translation.odoo-community.org/projects/server-ux-11-0/server-ux-11-0-sequence_check_digit/es/

LoisRForgeFlow committed:
[11.0][IMP] base_tier_validation: enhance views and register who does and who asks for reviews in new fields 'done_by' and 'requested_by'.

AdriaGForgeFlow committed:
[11.0][FIX] base_tier_validation

AdriaGForgeFlow committed:
Add name attribute to filter

Commit:
[MIG] base_tier_validation: Migration to 12.0

AdriaGForgeFlow committed:
[11.0][IMP] base_tier_validation

fixup and extend tests

[ADD] systray icon for pending reviews

[FIX] Remove python safe_eval

[ADD] base_tier_validation_formula and migration scripts

[ADD] widget domain and python expression to define reviewer in tier definition

[ADD] auto updating of systray icon counter

[ADD] validation date field

[ADD] review widget dropdown menu

etobella committed:
[IMP] base_tier_validation: Pass parameters as functions

etobella committed:
[IMP] base_tier_validation: Add a chatter message if the model has it

LoisRForgeFlow committed:
[11.0][FIX] base_tier_validation: make name translatable and display reviews' name and state correctly translated.

LoisRForgeFlow committed:
[11.0][UPD] base_tier_validation: roadmap.
* using similar approach to activities has already benn addressed.
* add a new point explaining review tooltip improvement possibilities.

OCA-git-bot committed:
[ADD] icon.png

LoisRForgeFlow committed:
[12.0][MIG] base_tier_validation: adapt systray to v12.

LoisRForgeFlow committed:
Added translation using Weblate (Spanish)

AdriaGForgeFlow committed:
[12.0][FIX] dropdown style width and position

Commit:
[12.0][MIG] sequence_check_digit

AdriaGForgeFlow committed:
[12.0][IMP] base_tier_validation module

fmdl committed:
[FIX] base_tier_validation : singleton error

liweijie0812 committed:
Added translation using Weblate (Chinese (Simplified))

digitalsatori committed:
Translated using Weblate (Chinese (Simplified))

Currently translated at 100.0% (59 of 59 strings)

Translation: server-ux-12.0/server-ux-12.0-base_tier_validation
Translate-URL: https://translation.odoo-community.org/projects/server-ux-12-0/server-ux-12-0-base_tier_validation/zh_CN/

LoisRForgeFlow committed:
[11.0][IMP] base_tier_validation: add an optional email
notification to possible reviewers.

LoisRForgeFlow committed:
[11.0][FIX] base_tier_validation: singleton issue

ps-tubtim committed:
[12.0][IMP] base_tier_validation

LoisRForgeFlow committed:
[12.0][FIX] base_tier_validation: message_subscribe_users removed in v12

pegonzalezspesol committed:
[IMP] base_tier_validation: black, isort

pegonzalezspesol committed:
[MIG] 13.0 base_tier_validation

ps-tubtim committed:
[12.0][fix] base_tier_validation

LoisRForgeFlow committed:
[13.0][UPD] base_tier_validation:

* fix test after the last forward-ported feature.
* add a descrition of new know issue.

LoisRForgeFlow committed:
[13.0][UPD] base_tier_validation: update copyrights information after rebranding of Eficent to ForgeFlow.

Commit:
Added translation using Weblate (Dutch)

jimhoefnagels committed:
[IMP] solve pre commit warnings

Commit:
[FIX] When accepting or rejecting a review followers should not be notified. A customer or supplier can be a follower of the object and should not receive internal messages about the tier validation.

oca-travis committed:
[UPD] Update sequence_check_digit.pot

OCA-git-bot committed:
[UPD] README.rst

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: server-ux-12.0/server-ux-12.0-sequence_check_digit
Translate-URL: https://translation.odoo-community.org/projects/server-ux-12-0/server-ux-12-0-sequence_check_digit/

etobella committed:
[12.0][FIX] base_tier_validation: Unlink reviews when unlink a validation record

dong-z committed:
Translated using Weblate (Chinese (Simplified))

Currently translated at 100.0% (74 of 74 strings)

Translation: server-ux-13.0/server-ux-13.0-base_tier_validation
Translate-URL: https://translation.odoo-community.org/projects/server-ux-13-0/server-ux-13-0-base_tier_validation/zh_CN/

Saran440 committed:
[12.0][FIX] base_tier_validation : convert datetime timezone

etobella committed:
[FIX] tier_validation: Fix usage of can_review on tier.review

etobella committed:
[FIX] tier_validation: Security issues with record permissions

sbidoul committed:
pre-commit update

eantones committed:
Translated using Weblate (Spanish)

Currently translated at 80.0% (60 of 75 strings)

Translation: server-ux-13.0/server-ux-13.0-base_tier_validation
Translate-URL: https://translation.odoo-community.org/projects/server-ux-13-0/server-ux-13-0-base_tier_validation/es/

sergio-teruel committed:
[FIX] base_tier_validation: All models are visible when create a new definition

Commit:
[IMP] base_tier_validation: notification hook

sergio-teruel committed:
[IMP] base_tier_validation: Make test more resilient with sales and purchase modules installed

sergio-teruel committed:
Update base_tier_validation/tests/test_tier_validation.py

Commit:
[IMP] sequence_check_digit: black, isort, prettier

Commit:
[13.0][MIG] sequence_check_digit

sergio-teruel committed:
[FIX] base_tier_validation: Remove test to avoid crash tests with other modules installed

kittiu committed:
[13.0][FIX] base_tier_validation, fix approve by sequence problem

- When use approve by sequence, still approve tier of the same reviewer
- When using approve_sequence option in any tier.definition there can be inconsistencies in the systray notifications

dong-z committed:
Translated using Weblate (Chinese (Simplified))

Currently translated at 96.1% (73 of 76 strings)

Translation: server-ux-13.0/server-ux-13.0-base_tier_validation
Translate-URL: https://translation.odoo-community.org/projects/server-ux-13-0/server-ux-13-0-base_tier_validation/zh_CN/

kittiu committed:
[IMP] : black, isort, prettier

kittiu committed:
[14.0][MIG] base_tier_validation

yostashiro committed:
[IMP] base_tier_validation: state check

This commit splits the state condition checking
part into another method, allowing inheriting
modules to override the conditions as needed.

LoisRForgeFlow committed:
[13.0][IMP] base_tier_validation: make module multi-company compatible.

kittiu committed:
Enhance definition's view, add Tabs

LoisRForgeFlow committed:
[14.0][UPD] base_tier_validation: update readme

etobella committed:
[IMP] base_tier_validation: Remove unnecessary modification of views

kittiu committed:
[FIX] fix test error, by separate test view into another class

oca-travis committed:
[UPD] Update sequence_check_digit.pot

etobella committed:
[FIX] tier_validation: Fix views from template as state was not used

kittiu committed:
[14.0][FIX] base_tier_validation, improve accuracy of systray

kittiu committed:
[14.0][IMP] base_tier_validation, add 'Todo by' column in review table

kittiu committed:
[14.0][FIX] base_tier_validation, access_token as exception field

LoisRForgeFlow committed:
[IMP] base_tier_validation: improve error message.

MiquelRForgeFlow committed:
[FIX] base_tier_validation: correct fusion of domains

anasuarez1 committed:
Translated using Weblate (Spanish)

Currently translated at 87.3% (76 of 87 strings)

Translation: server-ux-14.0/server-ux-14.0-base_tier_validation
Translate-URL: https://translation.odoo-community.org/projects/server-ux-14-0/server-ux-14-0-base_tier_validation/es/

schhatbar-initos committed:
[IMP] sequence_check_digit: black, isort, prettier

schhatbar-initos committed:
[14.0][MIG]sequence_check_digit

sergio-teruel committed:
[FIX] base_tier_validation: External users receive the internal notifications from requested tier validations
TT29001

JordiBForgeFlow committed:
[IMP] base_tier_validation: support to search for records where validation has not yet started

oca-travis committed:
[UPD] Update sequence_check_digit.pot

OCA-git-bot committed:
[UPD] README.rst

Yvesldff committed:
Added translation using Weblate (French)

JordiBForgeFlow committed:
[base_tier_validation][fix] Allow to search for validated=False and rejected = False

kittiu committed:
[14.0][IMP] base_tier_validation, use sudo to post message

JordiBForgeFlow committed:
[FIX] base_tier_validation: avoid singleton error

newtratip committed:
[14.0][ENH] Nootification message on restart tier

MateuGForgeFlow committed:
[14.0][IMP] base_tier_validation: add new review type

kittiu committed:
[14.0][IMP] base_tier_validation, add tier description on stripe

alessandro-fiorino committed:
Added translation using Weblate (Italian)

alan196 committed:
Added translation using Weblate (Spanish (Mexico))

AdriaGForgeFlow committed:
[13.0][IMP] base_tier_validation: review_user_count perfromance

kittiu committed:
base_tier_validation, approve by sequence with option to bypass for same reviewer

kittiu committed:
[14.0][ENH] base_tier_validation, add configuration

simonstromb committed:
Added translation using Weblate (Swedish)

rousseldenis committed:
[14.0][IMP] base_tier_validation: Add hook for state transition

JasminSForgeFlow committed:
[IMP] base_tier_validation: black, isort, prettier

JasminSForgeFlow committed:
[MIG] base_tier_validation: Migration to 15.0

dreispt committed:
[FIX] base_tier_validation: error on reviews after uninstalling Tier Validation support

Shide committed:
[14.0] [FIX] external_dependency python-stdnum

Fix external dependency from `stdnum` to `python-stdnum`

OCA-git-bot committed:
sequence_check_digit 14.0.1.0.1

Shide committed:
[FIX] Removed external dependencies already defined in Odoo base

OCA-git-bot committed:
sequence_check_digit 14.0.1.0.2

pedrobaeza committed:
[FIX+IMP] base_tier_validation: Don't mess searches with NewId + non saved data

You can't search for a NewId in a domain, and doing a search requires
to flush cache contents to DB for being effective, so this disregards
the correct use of this on the fly computed field.

Fortunately, we have a tool at ORM level for applying domains on cached
content and without worrying about NewId, which is filtered_domain, so
let's apply it.

OCA-git-bot committed:
base_tier_validation 15.0.1.0.1

JordiBForgeFlow committed:
[IMP] base_tier_validation: introduce new fields for validation messages.
- can_review_message
- validated_message
- rejected_message

This introduces the possibility to customize the messages.

OCA-git-bot committed:
base_tier_validation 15.0.1.1.0

CarlosRoca13 committed:
[FIX] base_tier_validation: Icon error when validated

OCA-git-bot committed:
base_tier_validation 15.0.1.2.0

bizzappdev committed:
[MIG]: base_tier_validation: Migration to 16

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 77.1% (81 of 105 strings)

Translation: server-ux-16.0/server-ux-16.0-base_tier_validation
Translate-URL: https://translation.odoo-community.org/projects/server-ux-16-0/server-ux-16-0-base_tier_validation/it/

ilyasProgrammer committed:
[IMP] base_tier_validation: group_erp_manager can edit tier definitions.

tate11 committed:
Added translation using Weblate (Turkish)

CRogos committed:
[FIX] pre-commit warnings

OCA-git-bot committed:
base_tier_validation 16.0.1.0.1

ilyasProgrammer committed:
base_tier_validation: skip_validation_check

mymage committed:
Translated using Weblate (Italian)

Currently translated at 77.1% (81 of 105 strings)

Translation: server-ux-16.0/server-ux-16.0-base_tier_validation
Translate-URL: https://translation.odoo-community.org/projects/server-ux-16-0/server-ux-16-0-base_tier_validation/it/

etobella committed:
base_tier_validation: Use domain for filtering values

OCA-git-bot committed:
base_tier_validation 16.0.1.0.2

etobella committed:
[FIX] base_tier_validation: Detect if model has active field

OCA-git-bot committed:
base_tier_validation 16.0.1.0.3

OCA-git-bot committed:
base_tier_validation 16.0.1.0.4

victoralmau committed:
[IMP] base_tier_validation: Misc changes

- Add _check_state_from_condition() method to better clarify the code and to be able to inherit it by another module.
- Add _add_tier_validation_x() methods to better clarify the code and to be able to inherit it by another module.
- Add name to th + td from ReviewsTable for a better selector if another module needs it.

TT43351

mymage committed:
Translated using Weblate (Italian)

Currently translated at 77.1% (81 of 105 strings)

Translation: server-ux-16.0/server-ux-16.0-base_tier_validation
Translate-URL: https://translation.odoo-community.org/projects/server-ux-16-0/server-ux-16-0-base_tier_validation/it/

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 77.1% (81 of 105 strings)

Translation: server-ux-16.0/server-ux-16.0-base_tier_validation
Translate-URL: https://translation.odoo-community.org/projects/server-ux-16-0/server-ux-16-0-base_tier_validation/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 77.1% (81 of 105 strings)

Translation: server-ux-16.0/server-ux-16.0-base_tier_validation
Translate-URL: https://translation.odoo-community.org/projects/server-ux-16-0/server-ux-16-0-base_tier_validation/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 77.1% (81 of 105 strings)

Translation: server-ux-16.0/server-ux-16.0-base_tier_validation
Translate-URL: https://translation.odoo-community.org/projects/server-ux-16-0/server-ux-16-0-base_tier_validation/it/

OCA-git-bot committed:
base_tier_validation 16.0.1.0.5

Kev-Roche committed:
[14.0][IMP] add write possibility for reviewers

mathben committed:
base_tier_validation: multiple comment with approve_sequence_bypass

- when option approve_sequence_bypass is enable and comments option with
sequence, the user add a comment and all level approbation add this
comment.
This code add comment only for appropriate tier validation.

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (104 of 104 strings)

Translation: server-ux-16.0/server-ux-16.0-base_tier_validation
Translate-URL: https://translation.odoo-community.org/projects/server-ux-16-0/server-ux-16-0-base_tier_validation/es/

hbrunn committed:
[ADD] set all form fields readonly when under validation

victoralmau committed:
[IMP] base_tier_validation: Add method to allow remove reviews.

A method for deciding whether or not to remove related reviews is added.

TT43351

alan196 committed:
[IMP] base_tier_validation: show in res.config.settings already migrated modules

OCA-git-bot committed:
base_tier_validation 16.0.1.0.6

fmdl committed:
[FIX] base_tier_validation: field is edtitable

And some part of the field are not show.

OCA-git-bot committed:
oca-github-bot post-merge updates

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (109 of 109 strings)

Translation: server-ux-16.0/server-ux-16.0-base_tier_validation
Translate-URL: https://translation.odoo-community.org/projects/server-ux-16-0/server-ux-16-0-base_tier_validation/es/

OpenCode committed:
[FIX] base_tier_validation: Exclude cancel records from pending count

OCA-git-bot committed:
[BOT] post-merge updates

victoralmau committed:
[IMP] base_tier_validation: Add validation_status field to tier.validation

TT33369

houzefa-abba committed:
Translated using Weblate (French)

Currently translated at 100.0% (109 of 109 strings)

Translation: server-ux-16.0/server-ux-16.0-base_tier_validation
Translate-URL: https://translation.odoo-community.org/projects/server-ux-16-0/server-ux-16-0-base_tier_validation/fr/

bosd committed:
[IMP] base_tier_validation: update tests to support messages

bosd committed:
[IMP] base_tier_validation: use sudo to post message

bosd committed:
base_tier_validation: only post notifications to reciepients

victoralmau committed:
[IMP] base_tier_validation: Add tier.validation.tester models to _get_tier_validation_model_names()

Even if they are models that are only used in tests, it is necessary to add the tier.validation.tester
and tier.validation.tester2 to the tier.definition's _get_tier_validation_model_names() method.

TT33369

OCA-git-bot committed:
[BOT] post-merge updates

celm1990 committed:
[PERF] base_tier_validation: filter records with reviews before compute fields
Before this commit, in a database with many records candidates for applying reviews, fields with a defined search function were very slow because the search was performed over all records with these states.
After this commit, first filter only records with review_ids set, which significantly reduces performance by more than 10X!!!

mymage committed:
Translated using Weblate (Italian)

Currently translated at 72.9% (81 of 111 strings)

Translation: server-ux-16.0/server-ux-16.0-base_tier_validation
Translate-URL: https://translation.odoo-community.org/projects/server-ux-16-0/server-ux-16-0-base_tier_validation/it/

OCA-git-bot committed:
[BOT] post-merge updates

celm1990 committed:
[IMP] base_tier_validation: filter tier definition for the current company
In a multicompany environment as follows:
User Request. Selected Companies: C1, C2, Current Company C1
User Approver: Selected Companies: C1 Current Company C1
Before this commit, when a user requested validation with multicompany enabled (C1, C2), definitions of other companies could be applied. If the user approver had access to fewer companies (C1) and confirmed the document, then the remaining reviews (C2) were ignored in the pending state. These reviews were shown on the sidebar but could never be approved because the document had already been approved.
After this commit, tier definitions are filtered only by the current company of the user request.

mymage committed:
Translated using Weblate (Italian)

Currently translated at 72.9% (81 of 111 strings)

Translation: server-ux-16.0/server-ux-16.0-base_tier_validation
Translate-URL: https://translation.odoo-community.org/projects/server-ux-16-0/server-ux-16-0-base_tier_validation/it/

OCA-git-bot committed:
[BOT] post-merge updates

victoralmau committed:
[IMP] base_tier_validation: Add _prepare_tier_review_vals() methot to allow inheritance

TT45804

OCA-git-bot committed:
[BOT] post-merge updates

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (114 of 114 strings)

Translation: server-ux-16.0/server-ux-16.0-base_tier_validation
Translate-URL: https://translation.odoo-community.org/projects/server-ux-16-0/server-ux-16-0-base_tier_validation/es/

OCA-git-bot committed:
[BOT] post-merge updates

bosd committed:
[IMP] base_tier_validation: test performance improvement

- Include context keys for avoiding mail operations overhead.

OCA-git-bot committed:
[BOT] post-merge updates

OCA-git-bot committed:
[BOT] post-merge updates

rven committed:
[IMP] base_tier_validation: pre-commit stuff

rven committed:
[MIG] base_tier_validation: Migration to 17.0

Commit:
[MIG] base_tier_validation_formula: Migrate to version 17.0

OCA-git-bot committed:
[BOT] post-merge updates

bosd committed:
[IMP] base_tier_validation: pre-commit stuff

OCA-git-bot committed:
[BOT] post-merge updates

bosd committed:
[IMP] base_tier_validation: Merge with base_tier_validation_waiting

bosd committed:
[IMP] base_tier_validation: review comments

Co-authored-by: Carlos Lopez <celm1990@hotmail.com>

bosd committed:
[IMP]base_tier_validation: UI Improvements

evanomnisoft committed:
Incorrect displayed timezone for tier review validation date

evanomnisoft committed:
Update CONTRIBUTORS.rst

victoralmau committed:
[FIX] base_tier_validation: Change web_ribbon text to title to make it translatable

OCA-git-bot committed:
[BOT] post-merge updates

OCA-git-bot committed:
[BOT] post-merge updates

evanomnisoft committed:
Renamed variables for logical consistentcy

Variables were inconsistently named leading to confusion when understanding the code.

bosd committed:
[RFC]base_tier_validation: remove unused code

OCA-git-bot committed:
[BOT] post-merge updates

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (118 of 118 strings)

Translation: server-ux-17.0/server-ux-17.0-base_tier_validation
Translate-URL: https://translation.odoo-community.org/projects/server-ux-17-0/server-ux-17-0-base_tier_validation/es/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 66.1% (78 of 118 strings)

Translation: server-ux-17.0/server-ux-17.0-base_tier_validation
Translate-URL: https://translation.odoo-community.org/projects/server-ux-17-0/server-ux-17-0-base_tier_validation/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 69.4% (82 of 118 strings)

Translation: server-ux-17.0/server-ux-17.0-base_tier_validation
Translate-URL: https://translation.odoo-community.org/projects/server-ux-17-0/server-ux-17-0-base_tier_validation/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (118 of 118 strings)

Translation: server-ux-17.0/server-ux-17.0-base_tier_validation
Translate-URL: https://translation.odoo-community.org/projects/server-ux-17-0/server-ux-17-0-base_tier_validation/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (118 of 118 strings)

Translation: server-ux-17.0/server-ux-17.0-base_tier_validation
Translate-URL: https://translation.odoo-community.org/projects/server-ux-17-0/server-ux-17-0-base_tier_validation/it/

bosd committed:
[FIX] base_tier_validation: Do not write review fields in pending state

bosd committed:
[IMP] base_tier_validation: Test created notification body

OCA-git-bot committed:
[BOT] post-merge updates

OCA-git-bot committed:
[BOT] post-merge updates

bosd committed:
Translated using Weblate (Dutch (nl_NL))

Currently translated at 94.9% (112 of 118 strings)

Translation: server-ux-17.0/server-ux-17.0-base_tier_validation
Translate-URL: https://translation.odoo-community.org/projects/server-ux-17-0/server-ux-17-0-base_tier_validation/nl_NL/

fmdl committed:
[FIX] base_tier_validation: prevent useless loop

manuelregidor committed:
[17.0] [IMP] base_tier_validation: Block notifications

bizzappdev committed:
[FIX][17.0]base_tier_validation: Fixed selection not translated.

OCA-git-bot committed:
[BOT] post-merge updates

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (119 of 119 strings)

Translation: server-ux-17.0/server-ux-17.0-base_tier_validation
Translate-URL: https://translation.odoo-community.org/projects/server-ux-17-0/server-ux-17-0-base_tier_validation/it/

houzefa-abba committed:
[FIX] base_tier_validation: Field merge in view

Previous implementation was losing fields in some cases; new one
properly merges dicts/tuples.

This fixes odd "Missing field string information" errors in
form-embedded lists.

This cset also removes an obsolete `base_model_name` context key no
longer used in Odoo 16.

This cset adds a `test_get_view` test which checks for that
`need_validation` field; it was failing with the previous impl.

Fixes #825.

OCA-git-bot committed:
[BOT] post-merge updates

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (125 of 125 strings)

Translation: server-ux-17.0/server-ux-17.0-base_tier_validation
Translate-URL: https://translation.odoo-community.org/projects/server-ux-17-0/server-ux-17-0-base_tier_validation/it/

OCA-git-bot committed:
[BOT] post-merge updates

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (125 of 125 strings)

Translation: server-ux-17.0/server-ux-17.0-base_tier_validation
Translate-URL: https://translation.odoo-community.org/projects/server-ux-17-0/server-ux-17-0-base_tier_validation/es/

TDu committed:
[IMP] base_tier_validation: Add reminder option

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (125 of 125 strings)

Translation: server-ux-17.0/server-ux-17.0-base_tier_validation
Translate-URL: https://translation.odoo-community.org/projects/server-ux-17-0/server-ux-17-0-base_tier_validation/it/

Kev-Roche committed:
[IMP] configurable domain for readonly fields in validation

jakobkrabbe committed:
Translated using Weblate (Swedish)

Currently translated at 100.0% (125 of 125 strings)

Translation: server-ux-17.0/server-ux-17.0-base_tier_validation
Translate-URL: https://translation.odoo-community.org/projects/server-ux-17-0/server-ux-17-0-base_tier_validation/sv/

ChrisOForgeFlow committed:
[16.0][FIX] base_tier_validation, search only active definitions

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (125 of 125 strings)

Translation: server-ux-17.0/server-ux-17.0-base_tier_validation
Translate-URL: https://translation.odoo-community.org/projects/server-ux-17-0/server-ux-17-0-base_tier_validation/it/

Shide committed:
[FIX] base_tier_validation: Tier Validation Exceptions

Shide committed:
[FIX] base_tier_validation: Invalidate review model on tests

Shide committed:
[FIX] base_tier_validation: Tier Validation Exception Groups

ChrisOForgeFlow committed:
[16.0][FIX] base_tier_validation, add sudo call to models without explicit access

Shide committed:
[FIX] base_tier_validation: Fix how readonly modifiers are built

OCA-git-bot committed:
[BOT] post-merge updates

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (141 of 141 strings)

Translation: server-ux-17.0/server-ux-17.0-base_tier_validation
Translate-URL: https://translation.odoo-community.org/projects/server-ux-17-0/server-ux-17-0-base_tier_validation/it/

OCA-git-bot committed:
[BOT] post-merge updates

bosd committed:
[IMP] base_tier_validation: label

Silence the warning: Two fields (model_name, model_id) of tier.validation.exception() have the same label: Model. [Modules: base_tier_validation and base_tier_validation]

OCA-git-bot committed:
[BOT] post-merge updates

OCA-git-bot committed:
[BOT] post-merge updates

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (143 of 143 strings)

Translation: server-ux-17.0/server-ux-17.0-base_tier_validation
Translate-URL: https://translation.odoo-community.org/projects/server-ux-17-0/server-ux-17-0-base_tier_validation/it/

OCA-git-bot committed:
[BOT] post-merge updates

OCA-git-bot committed:
[BOT] post-merge updates

StefanRijnhart committed:
[IMP] base_tier_validation: support computed state field

Computed fields bypass `write`, so we need to override `_write` for that case.
Also, the current value before the update needs to be fetched from the database
because the new value is already set in the cache.

OCA-git-bot committed:
[BOT] post-merge updates

LoisRForgeFlow committed:
[FIX] base_tier_validation: amend authors

I just realized that a new author was added to the module during an
improvement that merged the functionality of base_tier_validation_waiting.
This was overlook during my review and not explicitly accepted. There
have been many contributors adding new functionalities and this one
was not specially big to be included as author.

OCA-git-bot committed:
[BOT] post-merge updates

TDu committed:
Fix base_tier_validation reminder option

kevinkhao committed:
[IMP] base_tier_validation: pre-commit auto fixes

kevinkhao committed:
[IMP] base_tier_validation: odoo upgrade_code fixes
Co-authored-by: Do Anh Duy <duyda@trobz.com>

kevinkhao committed:
[MIG] base_tier_validation: Migration to 18.0
Co-authored-by: Do Anh Duy <duyda@trobz.com>

henrybackman committed:
[FIX] base_tier_validation: forward porting fixes

TierReview status "waiting" has been added and is the default. Modified the
tier review reminder search domain to accommodate
invalidate_cache() replaced with invalidate_recordset()

OCA-git-bot committed:
[BOT] post-merge updates

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (150 of 150 strings)

Translation: server-ux-17.0/server-ux-17.0-base_tier_validation
Translate-URL: https://translation.odoo-community.org/projects/server-ux-17-0/server-ux-17-0-base_tier_validation/it/

mymage committed:
Added translation using Weblate (Italian)

AlexPForgeFlow committed:
[IMP] base_tier_validation: add icon

OCA-git-bot committed:
[BOT] post-merge updates

etobella committed:
[MIG] sequence_check_digit: Migration to 16.0

yankinmax committed:
[IMP] sequence_check_digit: pre-commit auto fixes

yankinmax committed:
[MIG] sequence_check_digit: Migration to 18.0

rjaraspearhead commented:
@xaviedoanhduy 
Could this PR be directed to the 18.0 branch of OCA?

Your changes are better, and I could cancel my PR since we were both working on the same migration.

https://github.com/OCA/server-ux/pull/990

xaviedoanhduy commented:
> @xaviedoanhduy Could this PR be directed to the 18.0 branch of OCA?
> 
> Your changes are better, and I could cancel my PR since we were both working on the same migration.
> 
> [OCA#990](https://github.com/OCA/server-ux/pull/990)

hi @rjaraspearhead, I have created PR on OCA side (`https://github.com/OCA/server-ux/pull/992`), thank you for following up.

tarteo committed:
[FIX] base_technical_features: Add depends key in __manifest__.py

OCA-git-bot committed:
Merge PR #999 into 18.0

Signed-off-by pedrobaeza

OCA-git-bot committed:
[BOT] post-merge updates

OCA-git-bot committed:
Merge PR #988 into 18.0

Signed-off-by simahawk

Commit:
[UPD] Update sequence_check_digit.pot

OCA-git-bot committed:
[BOT] post-merge updates

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (70 of 70 strings)

Translation: server-ux-18.0/server-ux-18.0-date_range
Translate-URL: https://translation.odoo-community.org/projects/server-ux-18-0/server-ux-18-0-date_range/it/

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: server-ux-18.0/server-ux-18.0-sequence_check_digit
Translate-URL: https://translation.odoo-community.org/projects/server-ux-18-0/server-ux-18-0-sequence_check_digit/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (12 of 12 strings)

Translation: server-ux-18.0/server-ux-18.0-sequence_check_digit
Translate-URL: https://translation.odoo-community.org/projects/server-ux-18-0/server-ux-18-0-sequence_check_digit/it/

xaviedoanhduy committed:
[IMP] base_tier_validation: using _read_group for better performance
